docs/reference/filters.md

```diff
@@ -1084,6 +1084,99 @@ forwardToken("X-Tokeninfo-Forward")
 forwardToken("X-Tokeninfo-Forward", "access_token")
 ```
 
+## oauthGrant
+
+Enables authentication and authorization with an OAuth2 authorization code grant flow as
+specified by [RFC 6749 Section 1.3.1](https://tools.ietf.org/html/rfc6749#section-1.3.1).
+Automatically redirects unauthenticated users to log in at their provider's authorization
+endpoint. Supports token refreshing and stores access and refresh tokens in an encrypted
+cookie. Supports credential rotation for the OAuth2 client ID and secret.
+
+The filter consumes and drops the grant token request cookie to prevent it from leaking
+to untrusted downstream services.
+
+The filter must be used in conjuction with the [grantCallback](#grantCallback) filter
+where the OAuth2 provider can redirect authenticated users with an authorization code.
+
+The filter may be used with the [grantClaimsQuery](#grantClaimsQuery) filter to perform
+authz and access control.
+
+See the [tutorial](../tutorials/auth.md#oauth2-authorization-grant-flow) for step-by-step 
+instructions.
+
+Examples:
+
+```
+callback:
+    Path("/oauth/callback")
+    -> grantCallback()
+    -> <shunt>;
+
+all:
+    *
+    -> oauthGrant()
+    -> "http://localhost:9090";
+```
+
+Skipper arguments:
+
+| Argument | Required? | Description |
+| -------- | --------- | ----------- |
+| `-enable-oauth2-grant-flow` | **yes** | toggle flag to enable the `oauthGrant()` filter. Must be set if you use the filter in routes. Example: `-enable-oauth2-grant-flow` |
+| `-oauth2-auth-url` | **yes** | URL of the OAuth2 provider's authorize endpoint. Example: `-oauth2-auth-url=https://identity.example.com/oauth2/authorize` |
+| `-oauth2-token-url` | **yes** | URL of the OAuth2 provider's token endpoint. Example: `-oauth2-token-url=https://identity.example.com/oauth2/token` |
+| `-oauth2-tokeninfo-url` | **yes** | URL of the OAuth2 provider's tokeninfo endpoint. Example: `-oauth2-tokeninfo-url=https://identity.example.com/oauth2/tokeninfo` |
+| `-oauth2-callback-path` | **yes** | path of the Skipper route containing the `grantCallback()` filter for accepting an authorization code and using it to get an access token. Example: `-oauth2-callback-path=/oauth/callback` |
```

Comment:
docfix: i think the -oauth2-callback-path is optional. We use `/.well-known/oauth2-callback` by default.